### PR TITLE
Allow plugins to add custom header tags

### PR DESCRIPTION
**Pull request by keboeoris**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A

New hooks for inserting header tags via plugins

### CODE REVIEW

**Review by cedric-anne — COMMENTED** [review]
Could you please give some example of what HTML tasg you want to insert here?

Instead of having 3 different hooks, You could probably have a unique `add_header_tag` hook that expect to get a list of `{"tag": "meta", "properties": [{"name": "XXX", "content": "YYY"}]}` elements. It will probably be more flexible.

**Comment by cedric-anne** [comment]
Also, as it introduces a new feature, I changed the target branch to main.

File: src/Html.php [hunk 1]
```diff
@@ -1301,6 +1301,17 @@ public static function includeHeader(
...
         $tpl_vars['css_files'][] = ['path' => 'css/palettes/' . $theme . '.scss'];
 
+        // Add specific meta tags for plugins
+        if (isset($PLUGIN_HOOKS[Hooks::ADD_HEADER_TAG]) && count($PLUGIN_HOOKS[Hooks::ADD_HEADER_TAG])) {
+            foreach ($PLUGIN_HOOKS[Hooks::ADD_HEADER_TAG] as $plugin => $header_tags) {
+                if (!Plugin::isPluginActive($plugin)) {
+                    continue;
+                }
+
+                $tpl_vars['header_tags'] = $header_tags;
```
Comment (cedric-anne) [inline src/Html.php:1311]: If multiple plugins are using this hook, each will override the value defined by the preceding plugin. You should merge all the values returned in `$tpl_vars['header_tags']`.

File: templates/layout/parts/head.html.twig [hunk 2]
```diff
@@ -48,6 +48,13 @@
    {# prevent robots to index GLPI instances #}
    <meta name="robots" content="noindex, nofollow" />
 
+   {#  Custom header tags for plugins  #}
+   {% for header_tag in header_tags %}
+     {% for propertie in header_tag['properties'] %}
+        <{{ header_tag['tag'] }} {% for key, value in propertie %}{{ key }}="{{ value | e }}" {% endfor %}/>
```
Comment (cedric-anne) [inline templates/layout/parts/head.html.twig:54]: Escaping is done automatically.
```suggestion
        <{{ header_tag['tag'] }} {% for key, value in propertie %}{{ key }}="{{ value }}" {% endfor %}/>
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: src/Html.php (hunk 1 above)
Comment (keboeoris) [reply src/Html.php]: Thanks, We updated the related code.

File: templates/layout/parts/head.html.twig [hunk 3]
```diff
@@ -48,6 +48,14 @@
    {# prevent robots to index GLPI instances #}
    <meta name="robots" content="noindex, nofollow" />
 
+    {# Custom header tags for plugins #}
+    {% for header_tag in header_tags %}
+        {% for property in header_tag.properties %}
+            <{{ header_tag.tag }} {% for key, value in property %}{{ key }}="{{ value }}" {% endfor %}/>
+        {% endfor %}
```
Comment (cedric-anne) [inline templates/layout/parts/head.html.twig:55]: I think the correct version is this one:
```suggestion
        <{{ header_tag.tag }} {% for key, value in header_tag.properties %}{{ key }}="{{ value }}" {% endfor %}/>
```

Comment (keboeoris) [reply templates/layout/parts/head.html.twig]: Indeed, we were using an array of tags in the hook call.

**Review by cedric-anne — APPROVED** [review]
Validated using https://github.com/pluginsGLPI/example/pull/54

**Review by AdrienClairembault — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**